presto-cassandra/src/main/java/io/prestosql/plugin/cassandra/NativeCassandraSession.java

```diff
@@ -220,24 +221,24 @@ public CassandraTable getTable(SchemaTableName schemaTableName)
         for (ColumnMetadata columnMeta : tableMeta.getPartitionKey()) {
             primaryKeySet.add(columnMeta.getName());
             boolean hidden = hiddenColumns.contains(columnMeta.getName());
-            CassandraColumnHandle columnHandle = buildColumnHandle(tableMeta, columnMeta, true, false, columnNames.indexOf(columnMeta.getName()), hidden);
-            columnHandles.add(columnHandle);
+            Optional<CassandraColumnHandle> columnHandle = buildColumnHandle(tableMeta, columnMeta, true, false, columnNames.indexOf(columnMeta.getName()), hidden);
+            columnHandle.ifPresent(columnHandles::add);
```

Comment:
Is it safe to ignore unsupported primary key columns?
For select it might work. What will happen for insert?

Reply:
When we ignore primary key, the query both select and insert fail with below message.

`SELECT DISTINCT queries must only request partition key columns and/or static columns (not c1)`

```
cqlsh> create table test.test_tuple3 (c1 int, c2 tuple<int>, c3 int, c4 set<frozen<tuple<int>>>, primary key (c2));

presto> insert into test.test_tuple3 select 1, 2;
Query 20190408_111315_00006_2cmik failed: SELECT DISTINCT queries must only request partition key columns and/or static columns (not c1)

presto> select * from test.test_tuple3;
Query 20190408_111531_00007_2cmik failed: SELECT DISTINCT queries must only request partition key columns and/or static columns (not c1)
```

Reply:
If this fails on unsupported part key column data type, we should generate explicit err for such a case.
And be graceful for ordinary columns.

Reply:
eg
```
CassandraColumnHandle = columnHandle = buildColumnHandle(....)
    .orElseThrow(() -> throw new PrestoException(... )
columnHandles.add(columnHandle);
```